File: server/php/src/Presentation/Models/GetPaymentDetails/Response.php
<?php

namespace OnlinePayments\ExampleApp\Presentation\Models\GetPaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\HostedCheckoutSpecificOutput;
use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\OperationOutput;
use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\PaymentOutput;
use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\PaymentStatusOutput;
use OnlinePayments\ExampleApp\Application\Interfaces\Responses\ResponseInterface;

class Response implements ResponseInterface
{
    /** @var OperationOutput[]|null */
    public ?array $operations = null;

    public ?HostedCheckoutSpecificOutput $hostedCheckoutSpecificOutput = null;

    public ?PaymentOutput $paymentOutput = null;

    public ?string $status = null;

    public ?PaymentStatusOutput $statusOutput = null;

    public ?string $id = null;

    public function toArray(): array
    {
        return [
            'operations' => $this->operations !== null
                ? array_map(
                    fn (OperationOutput $op) => $op->toArray(),
                    $this->operations
                )
                : [],
            'hostedCheckoutSpecificOutput' => ($this->hostedCheckoutSpecificOutput ?? new HostedCheckoutSpecificOutput())->toArray(),
            'paymentOutput' => ($this->paymentOutput ?? new PaymentOutput())->toArray(),
            'status' => $this->status,
            'statusOutput' => ($this->statusOutput ?? new PaymentStatusOutput())->toArray(),
            'id' => $this->id,
        ];
    }
}

File: server/python/app/presentation/models/get_payment_details/response.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from pydantic.alias_generators import to_camel
from app.application.domain.payments.payment_details.hosted_checkout_specific_output import HostedCheckoutSpecificOutput
from app.application.domain.payments.payment_details.operation_output import OperationOutput
from app.application.domain.payments.payment_details.payment_output import PaymentOutput
from app.application.domain.payments.payment_details.payment_status_output import PaymentStatusOutput

class GetPaymentDetailsResponse(BaseModel):
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)

    id: Optional[str] = None
    status: Optional[str] = None
    status_output: Optional[PaymentStatusOutput] = None
    payment_output: Optional[PaymentOutput] = None
    hosted_checkout_specific_output: Optional[HostedCheckoutSpecificOutput] = None
    operations: Optional[List[OperationOutput]] = None